### PR TITLE
Resolve KubeArmor Installation Issue: Transition to Helm-Based Method in MicroK8s Community Addons

### PR DESCRIPTION
**Pull request by udit-uniyal**
This PR addresses an issue with the previous installation method of KubeArmor within the microk8s-community-addons/addons/kubearmor directory. The prior installation and removal methods were causing unsuccessful KubeArmor installations within MicroK8s.

### **Details**

**Issue Resolved:** The issue causing unsuccessful KubeArmor installation within MicroK8s due to outdated installation and removal methods has been resolved in this PR.

**Updated Approach:** The outdated installation and removal methods have been replaced entirely with a Helm-based approach.

**Helm Installation:** Helm is now responsible for installing the latest version of KubeArmor into MicroK8s.

Comprehensive Testing: The entire installation process, from installing to removing KubeArmor within MicroK8s, has been thoroughly tested to ensure functionality.

### **File Modifications**

Changes have been made to the enable and disable files to accommodate the new Helm-based installation and removal methods for KubeArmor within the specified directory.

This update ensures a more reliable and successful KubeArmor installation process within MicroK8s by adopting Helm as the installation method.

### CODE REVIEW

**Comment by ktsakalozos** [comment]
Thank you for this PR @udit-uniyal could you please sign the CLA so we can distribute the addon with MicroK8s? The error we are having is:
```
Checking the following users on Launchpad:
- udituniyal71@gmail.com ✕ (has no Launchpad account)
```

We have to install a specific version of kubearmor  via this addon so we do not break future releases and we offer same user experience within a K8s release.

I assume you are in sync with @gopiak regarding this change.

**Comment by udit-uniyal** [comment]
Hi @ktsakalozos, I have signed the CLA. You can distribute the addon now.

File: addons/kubearmor/disable [hunk 1]
```diff
@@ -1,8 +1,9 @@
 #!/usr/bin/env bash
 
+
 echo "Removing kubearmor from k8s cluster"
 
-sudo microk8s karmor uninstall
+microk8s.helm uninstall kubearmor-operator -n kubearmor
```
Comment (ktsakalozos) [inline addons/kubearmor/disable:6]: Please call the helm command as shown in https://github.com/canonical/microk8s-community-addons/blob/main/addons/cilium/enable

**Review by ktsakalozos — CHANGES_REQUESTED** [review]
Please see the comment on the helm call. Thank you.

File: addons/kubearmor/disable (hunk 1 above)
Comment (udit-uniyal) [reply addons/kubearmor/disable]: @ktsakalozos Could you please review the change now?

File: addons/kubearmor/enable [hunk 2]
```diff
@@ -2,12 +2,18 @@
...
 
-curl -sfL http://get.kubearmor.io/ | sudo sh -s -- -b "$SNAP_COMMON/bin"
+"$SNAP/microk8s-enable.wrapper" helm3
 
-cp "$CURRENT_DIR/karmor" "$SNAP_COMMON/plugins"
+echo "To view policy violation logs use karmor logs --gRPC 127.0.0.1:32767"
+
+${SNAP}/microk8s-helm3.wrapper repo add kubearmor https://kubearmor.github.io/charts
+${SNAP}/microk8s-helm3.wrapper repo update kubearmor
+${SNAP}/microk8s-helm3.wrapper upgrade --install kubearmor-operator kubearmor/kubearmor-operator -n kubearmor --set autoDeploy=true --create-namespace
```
[the remark this answers is not in the document]
Comment (udit-uniyal) [reply addons/kubearmor/enable]: Hi @ktsakalozos I have updated the file to specify the particular version that should be deployed during installation

[the remark this answers is not in the document]
Comment (udit-uniyal) [reply addons/kubearmor/enable]: Hi @ktsakalozos Any update?

**Comment by ktsakalozos** [comment]
Thank you for this work @udit-uniyal It will be available with the next 1.29 patch release.

**Pull request merged**